internal/services/executor/executor.go

```diff
@@ -1517,3 +1531,50 @@ func (e *Executor) Run(ctx context.Context) error {
 
 	return nil
 }
+
+func (e *Executor) tasksTimeoutCleanerLoop(ctx context.Context) {
+	for {
+		e.log.Debug().Msgf("tasksTimeoutCleaner")
+
+		e.tasksTimeoutCleaner(ctx)
+
+		sleepCh := time.NewTimer(2 * time.Second).C
+		select {
+		case <-ctx.Done():
+			return
+		case <-sleepCh:
+		}
+	}
+}
+
+func (e *Executor) tasksTimeoutCleaner(ctx context.Context) {
+	for _, rtID := range e.runningTasks.ids() {
+		if rt, ok := e.runningTasks.get(rtID); ok {
+			rt.Lock()
+			if rt.et.Status.Phase == types.ExecutorTaskPhaseRunning && rt.et.Spec.TaskTimeoutInterval != 0 && time.Since(*rt.et.Status.StartTime) >= rt.et.Spec.TaskTimeoutInterval {
+				rt.et.Status.Phase = types.ExecutorTaskPhaseTimedout
+				rt.et.Status.EndTime = util.TimeP(time.Now())
+				rt.et.Status.FailError = "timedout"
```

Comment:
FailError is probably old and not used except on the scheduler in one code path. I'll avoid setting it in the executor and review it's usage in future.